lib/common.bash

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This file contains common functions that
+# are being used by our metrics and integration tests
+
+die(){
+	msg="$*"
+	echo "ERROR: $msg" >&2
+	exit 1
+}
+
+# Gets versions and paths of all the components
+# list in kata-env
+extract_kata_env(){
+	local toml
+
+	toml="$(kata-runtime kata-env)"
+
+	# The runtime path itself, for kata-runtime, will be contained in the `kata-env`
+	# section. For other runtimes we do not know where the runtime Docker is using lives.
+	RUNTIME_CONFIG_PATH=$(awk '/^\[Runtime\]$/ {foundit=1} /^    Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+	RUNTIME_VERSION=$(awk '/^  \[Runtime.Version\]$/ {foundit=1} /^    Semver =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+	RUNTIME_COMMIT=$(awk '/^  \[Runtime.Version\]$/ {foundit=1} /^    Commit =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+
+	SHIM_PATH=$(awk '/^\[Shim\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+	SHIM_VERSION=$(awk '/^\[Shim\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {$1=$2=""; print $0; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+
+	PROXY_PATH=$(awk '/^\[Proxy\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+	PROXY_VERSION=$(awk '/^\[Proxy\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {print $5; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+
+	HYPERVISOR_PATH=$(awk '/^\[Hypervisor\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+	HYPERVISOR_VERSION=$(awk '/^\[Hypervisor\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {$1=$2=""; print $0; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+
+	INITRD_PATH=$(awk '/^\[Initrd\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+}
+
+# Checks that processes are not running
+check_processes() {
+	extract_kata_env
+	# Get system architecture
+	arch=$(go env GOARCH)
+	if [ "$arch" == "arm64" ]; then
```

Comment:
I don't remember seeing this before. Do we know why we do this here? Given we are checking there are no kata processes laying around, I thought this check would be pretty `arch` neutral?

Reply:
@grahamwhaley , yeah I got confused by this https://github.com/kata-containers/tests/blob/master/.ci/aarch64/clean_up_aarch64.sh#L27, I will modified to be neutral and in for the `clean_up_aarch64.sh` create another function to check `docker-containerd-shim`, what do you think? thanks

Reply:
Ah, that is where it came from. Right. yeah, you can probably just open code the pgrep over in the aarch64 code in place of the call to check_processes - or, you could make a 'check_process' that takes a single argument (like check_processes used to), and then call that from both the aarch64 code and from check_processes - that is, factor out the pgrep/kill stuff :-). Either way works for me right now.